libraries/AP_Motors/AP_MotorsHeli_Dual.cpp

```diff
@@ -302,81 +302,61 @@ void AP_MotorsHeli_Dual::calculate_scalars()
-                swash_tilt = roll_input + _yaw_scaler * yaw_input;
-            } else if (swash_num == 2) {
-                swash_tilt = roll_input - _yaw_scaler * yaw_input;
-            }
-        } else if (swash_axis == AP_MOTORS_HELI_DUAL_SWASH_AXIS_PITCH) {
-        // pitch tilt
-            if (swash_num == 1) {
-                swash_tilt = 0.0f;
-            } else if (swash_num == 2) {
-                swash_tilt = 0.0f;
-            }
-        } else if (swash_axis == AP_MOTORS_HELI_DUAL_SWASH_AXIS_COLL) {
-        // collective
-            if (swash_num == 1) {
-                swash_tilt = 0.45f * _dcp_scaler * (pitch_input + constrain_float(_dcp_trim, -0.2f, 0.2f)) + coll_input;
-            } else if (swash_num == 2) {
-                swash_tilt = -0.45f * _dcp_scaler * (pitch_input + constrain_float(_dcp_trim, -0.2f, 0.2f)) + coll_input;
-            }
-        }
-    }
-    return swash_tilt;
+    // Differential cyclic roll is used for yaw and combined for roll
+    const float swash1_roll = roll_input + _yaw_scaler * yaw_input;
+    const float swash2_roll = roll_input - _yaw_scaler * yaw_input;
+
+    // cyclic is not used for pitch control
+    const float pitch = 0.0;
+
+    // Differential collective for pitch and combined for thrust
+    const float swash1_coll =  0.45 * _dcp_scaler * (pitch_input + constrain_float(_dcp_trim, -0.2, 0.2)) + collective1_input;
+    const float swash2_coll = -0.45 * _dcp_scaler * (pitch_input + constrain_float(_dcp_trim, -0.2, 0.2)) + collective2_input;
+
+    // Calculate servo positions in swashplate library
+    _swashplate1.calculate(swash1_roll, pitch, swash1_coll);
+    _swashplate2.calculate(swash2_roll, pitch, swash2_coll);
+}
+
+// Mix and output swashplates for transverse
+void AP_MotorsHeli_Dual::mix_transverse(float pitch_input, float roll_input, float yaw_input, float collective1_input, float collective2_input)
+{
+    // cyclic is not used for roll control
+    const float roll = 0.0;
+
+    // Differential cyclic pitch is used for yaw
+    const float swash1_pitch = pitch_input - _yaw_scaler * yaw_input;
+    const float swash2_pitch = pitch_input + _yaw_scaler * yaw_input;
+
+    // Differential collective for roll and combined for thrust
+    const float swash1_coll =  0.45 * _dcp_scaler * (roll_input + constrain_float(_dcp_trim, -0.2, 0.2)) + collective1_input;
+    const float swash2_coll = -0.45 * _dcp_scaler * (roll_input + constrain_float(_dcp_trim, -0.2, 0.2)) + collective2_input;
+
+    // Calculate servo positions in swashplate library
+    _swashplate1.calculate(roll, swash1_pitch, swash1_coll);
+    _swashplate2.calculate(roll, swash2_pitch, swash2_coll);
+}
+
+// Mix and output swashplates for intermeshing
+void AP_MotorsHeli_Dual::mix_intermeshing(float pitch_input, float roll_input, float yaw_input, float collective1_input, float collective2_input)
+{
+    // Direct roll control on both swash plates
+    const float roll = roll_input;
```

Comment:
same nit pick: can we call this swash_roll

Reply:
done